pkg/kubectl/cmd/clusterinfo_dump.go

```diff
@@ -194,19 +194,34 @@ func dumpClusterInfo(f cmdutil.Factory, cmd *cobra.Command, args []string, out i
 
 		for ix := range pods.Items {
 			pod := &pods.Items[ix]
+			containers := pod.Spec.Containers
+
 			writer := setupOutputWriter(cmd, out, path.Join(namespace, pod.Name, "logs.txt"))
-			writer.Write([]byte(fmt.Sprintf("==== START logs for %s/%s ====\n", pod.Namespace, pod.Name)))
-			request, err := f.LogsForObject(pod, &api.PodLogOptions{}, timeout)
-			if err != nil {
-				return err
-			}
 
-			data, err := request.DoRaw()
-			if err != nil {
-				return err
+			for i := range containers {
+				container := containers[i]
+				startMsg := fmt.Sprintf("==== START logs for container %s of pod %s/%s ====\n", container.Name, pod.Namespace, pod.Name)
+				endMsg := fmt.Sprintf("==== END logs for container %s of pod %s/%s ====\n", container.Name, pod.Namespace, pod.Name)
+
+				writer.Write([]byte(startMsg))
+				request, err := f.LogsForObject(pod, &api.PodLogOptions{Container: container.Name}, timeout)
+				if err != nil {
+					// Print error and carry on dumping other pods info.
+					writer.Write([]byte(fmt.Sprintf("Create log request error: %s\n", err.Error())))
+					writer.Write([]byte(endMsg))
+					continue
+				}
+
+				data, err := request.DoRaw()
+				if err != nil {
+					// Print error and carry on dumping other pods info.
+					writer.Write([]byte(fmt.Sprintf("Request log error: %s\n", err.Error())))
+					writer.Write([]byte(endMsg))
```

Comment:
How about turning this loop body into a function and use `defer` for printing that message? This will ensure we have it printed always, esp. when introducing new branches here.